opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -575,15 +586,8 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             with self._lock:
                 self.attributes[key] = value
 
+    @_check_span_ended
     def _add_event(self, event: EventBase) -> None:
-        with self._lock:
-            if not self.is_recording():
```

Comment:
We're removing the `is_recording` logic here?

Reply:
after looking at this, would there be a case where is_recording is useful to check here? wont a Span in the SDK always return `True`?

Reply:
Right as discussed, we can remove this logic for functions within `trace`.